tests/telemetry/events/host_events.py

```diff
@@ -89,3 +108,26 @@ def mask_container(duthost):
 
     duthost.shell("systemctl unmask {}".format(container))
     duthost.shell("systemctl restart {}".format(container))
+
+
+def kill_critical_process(duthost):
+    logger.info("Killing critical process for exited unexpectedly event")
+    pid, container = get_critical_process(duthost)
+    assert pid != "", "No available process for testing"
+
+    change_autorestart = False
+    autorestart = duthost.shell("show feature autorestart {}".format(container))['stdout_lines']
+    if "disabled" in str(autorestart):
+        change_autorestart = True
+        duthost.shell("config feature autorestart {} enabled".format(container))
+
+    duthost.shell("docker exec {} kill -9 {}".format(container, pid), module_ignore_errors=True)
+
+    # Wait until specified container is not running because of critical process exit
+    wait_until(30, 5, 0, is_container_down, duthost, container)
+
+    if change_autorestart:
+        duthost.shell("config feature autorestart {} disabled".format(container))
+
+    duthost.shell("systemctl reset-failed {}".format(container))
+    duthost.shell("systemctl restart {}".format(container))
```

Comment:
If container not fully restart, will next test failed? suggest wait fully restart here.